actionview/lib/action_view/template/resolver.rb

```diff
@@ -267,14 +264,18 @@ def build_unbound_template(template, virtual_path)
         )
       end
 
-      def reject_files_external_to_app(files)
-        files.reject { |filename| !inside_path?(@path, filename) }
-      end
+      # Safe glob within @path
+      def template_glob(glob)
+        query = File.join(escape_entry(@path), glob)
+        path_with_slash = File.join(@path, "")
 
-      def inside_path?(path, filename)
-        filename = File.expand_path(filename)
-        path = File.join(path, "")
-        filename.start_with?(path)
+        Dir.glob(query).reject do |filename|
```

Comment:
Any reason we can't reuse the `base:` parameter from the previous diff here? Does it return relative paths and we don't want it to, or?

Reply:
It returns a relative path, and it's better for us to have an absolute path so we can check `File.directory?`, and that the expanded path is within the expected folder.

Reply:
On actually, there's a method I missed removing in a rebase. This is also replacing the glob in `find_candidate_template_paths`, which did not use `base:`